drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -61,9 +64,14 @@ class Hmis::Hud::Client < Hmis::Hud::Base
   end
 
   scope :with_access, ->(user, *permissions, **kwargs) do
-    return none unless user.permissions?(*permissions, **kwargs)
+    pids = Hmis::Hud::Project.with_access(user, *permissions, **kwargs).pluck(:id)
+    base_scope = left_outer_joins(:projects, :wip, :data_source)
+
+    unenrolled_scope = user.permissions?(*permissions, **kwargs) ? unenrolled.left_outer_joins(:data_source).merge(GrdaWarehouse::DataSource.hmis(user)) : none
+    enrolled_scope = base_scope.where(p_t[:id].in(pids))
+    wip_scope = base_scope.where(wip_t[:project_id].in(pids))
 
-    joins(:data_source).merge(GrdaWarehouse::DataSource.hmis(user))
+    unenrolled_scope.or(enrolled_scope).or(wip_scope)
   end
```

Comment:
This seems to be working as intended, but I'm seeing a significant performance hit. My rails consoled timed out trying to evaluate this scope for a user that has access to a lot of projects.

I could think of two improvements, I see an impact but its pretty minor.
1) Get rid of some of the unnecessary Data source joins
2) Stop doing `left_outer_joins(:projects, :wip)` twice (once in this scope, once in "unenrolled" scope) - I have no idea if Rails would've been smart enough to collapse that already, I didn't look closely at the sql to compare

```
  scope :with_access, ->(user, *permissions, **kwargs) do
    pids = Hmis::Hud::Project.with_access(user, *permissions, **kwargs).pluck(:id)
    data_source_id = user.hmis_data_source_id
    base_scope = where(data_source_id: data_source_id).left_outer_joins(:projects, :wip)

    unenrolled_scope = user.permissions?(*permissions, **kwargs) ? base_scope.unenrolled : none
    enrolled_scope = base_scope.where(p_t[:id].in(pids))
    wip_scope = base_scope.where(wip_t[:project_id].in(pids))

    unenrolled_scope.or(enrolled_scope).or(wip_scope)
  end
```

@eanders do you have any other suggestion for speeding this up? I'm thinking we could cache something if needed, like project access list, but it's tricky since it's all based on permission.